app/code/Magento/Catalog/Model/CategoryLinkRepository.php

```diff
@@ -21,20 +26,28 @@ class CategoryLinkRepository implements \Magento\Catalog\Api\CategoryLinkReposit
      */
     protected $productRepository;
 
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
     /**
      * @param \Magento\Catalog\Api\CategoryRepositoryInterface $categoryRepository
      * @param \Magento\Catalog\Api\ProductRepositoryInterface $productRepository
+     * @param LoggerInterface|null $logger
      */
     public function __construct(
         \Magento\Catalog\Api\CategoryRepositoryInterface $categoryRepository,
-        \Magento\Catalog\Api\ProductRepositoryInterface $productRepository
+        \Magento\Catalog\Api\ProductRepositoryInterface $productRepository,
+        ?LoggerInterface $logger
```

Comment:
Must not be nullable and should be `null` by default.

Reply:
@orlangur is using nullable types prohibited?

Reply:
@p-bystritsky this is just not needed in constructors as there is no sense to pass `null` explicitly. You either use default `null` value or pass a correct object.